File: public/docs/_examples/user-input/ts/app/click-me2.component.ts
// #docregion
import {Component} from 'angular2/core';

@Component({
  selector: 'click-me2',
  template: `
    <button (click)="onClickMe2($event)">No! .. Click me!</button>
    {{clickMessage}}`
})
export class ClickMeComponent2 {
  clickMessage = '';
  clicks = 1;

  onClickMe2(event:any){
    let evtMsg = event ? ' Event target is '+ event.target.tagName  : '';
    this.clickMessage = (`Click #${this.clicks++}. ${evtMsg}`)
  }
}

File: public/docs/_examples/user-input/dart/lib/click_me2_component.dart
// #docregion
import 'package:angular2/core.dart';

@Component(
    selector: 'click-me2',
    template: '''
      <button (click)="onClickMe2(\$event)">No! .. Click me!</button>
      {{clickMessage}}''')
class ClickMe2Component {
  String clickMessage = '';
  int _clicks = 1;

  void onClickMe2(dynamic event) {
    var evtMsg =
        event != null ? ' Event target is ' + event.target.tagName : '';
    clickMessage = ('Click #${_clicks++}. ${evtMsg}');
  }
}
